Services/Badge/classes/class.ilBadgeManagementGUI.php

```diff
@@ -384,11 +386,25 @@ protected function saveBadge(): void
 
             $badge->create();
 
-            if ($form->getInput("img_mode") === "up") {
-                $badge->uploadImage($_FILES["img"]);
-            } else {
-                $tmpl = new ilBadgeImageTemplate($form->getInput("tmpl"));
-                $badge->importImage($tmpl->getImage(), $tmpl->getImagePath());
+
+
+            try {
+                if ($form->getInput("img_mode") === "up") {
+                    $badge->uploadImage($_FILES["img"]);
+                } else {
+                    $tmpl = new ilBadgeImageTemplate($form->getInput('tmpl'));
+                    $badge->importImage($tmpl->getImage(), $tmpl->getImagePath());
+                }
+            } catch (BadgeException $e) {
+                switch ($e->getCode()) {
+                    case BadgeException::EXCEPTION_FILE_NOT_FOUND:
+                        $this->tpl->setOnScreenMessage('failure', $lng->txt('badge_uploaded_image_file_not_found'), true);
+                        break;
+                    case BadgeException::EXCEPTION_MOVE_UPLOADED_IMAGE_FAILED:
+                        $this->tpl->setOnScreenMessage('failure', $lng->txt('badge_create_image_processing_failed'), true);
+                }
+                $badge->delete();
+                $ilCtrl->redirect($this, "listBadges");
```

Comment:
Please delete and redirect only in case one of the known exception codes is given.